CDCgov/prime-reportstream
# Implemented parallel downloads in batch step

This PR adds the use of coroutines in the batch step to facilitate parallel download and processing of report files.

Test Steps:

1. Set debug level logging in `src/main/resources/log4j2.xml` (line 31)
2. Run smoke tests on this feature branch
3. Observe logs for `Starting download for blobUrl` and corresponding `Finished` messages. You should see multiple downloads start in parallel.

## Changes
- BatchFunction.batchUniversalData() and WorkflowEngine.createHeader() now make use of `runBlocking`, `async`, and `awaitAll` to enable parallel download and processing of reportfiles.
- Added debug level logging for individual downloads; info level logging for aggregated duration of downloads

## Checklist

### Testing

## Linked Issues
- Fixes #issue

## To Be Done
- Logging
- Test instructions (in this PR)
